Feature/2022 05 25 cisse tsyen rtk stage v2 (#703)
ordina-jworks/ordina-jworks.github.io: _posts/2022-05-04-RTK-Intership.md

{: refdef}
### Calculation of the range
On a basic conceptual level, the range is calculated by determining the amount of carrier waves between the satellite and the rover, which is then multiplied by the length of the carrier.

In this range there are still errors, these errors are caused by satellite clock and ephemeris, but also by ionospheric and tropospheric delays. 
We can solve this by sending correction data from the base station to the rover over a mobile network or another type of wireless communication such as LoRa.
### Network RTK
Network RTK is based on the use of several fixed reference stations spread out over the network. 
Depending on the application, the rover sends its location to the reference stations, so the rover is not dependent on one base station but can connect to several reference stations. 
This allows for a greater range than a single fixed base station, which is about 10 km.
### How to receive correction data?
There are a few ways to receive this correction data. You can build your own base station or use a service. 

> Not all services are free, for example Skylark, this service has coverage over 3 continents but is paying. 

Reviewer: ```suggestion
Not all services are free.
For example Skylark, this service has coverage over 3 continents but it is a paid service. 
```

These correction streams make it possible to determine an accurate position anywhere in Flanders. 
Using FLEPOS services is completely free, the connection costs over mobile networks, such as 4G, are not free.
{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/FLEPOS.png' | prepend: site.baseurl }}" alt="FLEPOS" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

### Usage of FLEPOS
FLEPOS is used within various markets that require accurate positioning:

- Surveying, photogrammetry
- Precision agriculture
- GIS inventory
- Hydrography
- Machine tool control
- Dredging
- ...

A specific RTK service has been set up for each of these markets. 
FLEPOS also offers data for scientific research. 
Before FLEPOS can be used, the registration form must be completed and approved on the website of Digitaal Vlaanderen.

### FLEPOS mountpoint
To receive the correction data, it is necessary to connect to a mountpoint of FLEPOS. 
The figure below shows all available mountpoints. 
The most commonly used data streams are FLEPOSVRS31GR and FLEPOSVRS32GREC. 

> The user is free to take one of the other available data streams for the GNSS receiver. 

Reviewer: ```suggestion
The user is free to take one of the available data streams for the GNSS receiver. 
```

Data stream FLEPOSMARIVRS31GR is recommended only for measurement in the North Sea because this data stream uses the 2 overseas stations ALDB and SHOE and the station BLIG located on the Bligh sandbank.

{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/Mounpoints.png' | prepend: site.baseurl }}" alt="FLEPOS" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

> ## RTK module and antenna

Reviewer: Explain that you'll now go into the implementation of a system and why you choose this system.

### SparkFun GPS-RTK Board - NEO-M8P-2
The Sparkfun GPS-RTK Board is a powerful breakout board for the NEO-M8P-2 module from u-blox.
The NEO-M8P-2 is a highly accurate module for the use of GNSS and GPS location solutions such as RTK.
The module is unique in that it can be configured as a rover as well as a base station.
There are 4 different methods to connect with this board: USB, I²C, UART or SPI.

> I'm using a combination with I²C and USB.

Reviewer: ```suggestion
I'm using a combination of I²C and USB.
```

{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/RTKBoard.jpg' | prepend: site.baseurl }}" alt="RTKBoard" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

### Antenna

> First, I had a cheap antenna that was laying around at the office.

Reviewer: ```suggestion
First, I had a cheap antenna that was already available in the office.
```

With this cheap antenna the signals were weak, inaccurate and it took a long time to receive them.
Therefore the testing was quite challenging and hard to do.

> That's why a good, accurate antenna,  like the  ANN-MB-00 GNSS multiband antenna, is necessary.

Reviewer: ```suggestion
That's why a good, accurate antenna, like the  ANN-MB-00 GNSS multiband antenna, is necessary.
```

### ANN-MB-00 GNSS multiband antenna
The ANN-MB-00 GNSS multiband antenna is unique from other GNSS/GPS antennas in that it is designed to receive both the classic L1 GPS band and the fairly newly launched L2 GPS band. 
Furthermore, the u-blox ANN-MB-00 is well constructed with a magnetic base with mounting holes for additional anchoring for the toughest environments.
It is suitable for use with any GPS/GNSS receiver with dual L1/L2 reception and supports GPS, GLONASS, Galileo and BeiDou.
{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/antenna.jpg' | prepend: site.baseurl }}" alt="ANN-MB-00 antenna" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

### Testing of the antennas
I tested both antennas side by side for 5 minutes while outside on the table, using SBAS as the correction method.

> These are the results (the more compact the data points the more accurate):

Reviewer: ```suggestion
These are the results, the more compact the data points the more accurate:
```

{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/TestingAntennasOutside.JPG' | prepend: site.baseurl }}" alt="TestingAntennasOutside" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

u-blox ANN-MB-00-00 (new antenne):
{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/newantenna.png' | prepend: site.baseurl }}" alt="newantenna" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

You can see that the antenna stays within a range of max 2.5m when stationary. 
It got a connection to the signal immediately when I started the module.

Unknown first cheap antenna:
{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/Cheapantenna.png' | prepend: site.baseurl }}" alt="Cheapantenna" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

> First of all it took way longer to establish a connection, a couple of minutes. 

Reviewer: ```suggestion
First of all, it took multiple minutes to establish a connection. 
```

The data points are further apart from each other. 

> Whereas you can see with the other antenna that they are much closer together. 

Reviewer: ```suggestion
You can see that with the other antenna the data points are much closer together. 
```

So that means that there is a lot of play between each data point. 
This means that the new antenna is more accurate. 

> At last you can also see that the range of the antenna went over the 2.5 meter mark.

Reviewer: ```suggestion
Finally, you can also see that the range of the antenna went over the 2.5 meter mark.
```

From this we can conclude that it is necessary to have a good antenna when we want to work precisely.

### Testing with RTK

> There were a couple of problems with the SparkFun GPS-RTK Board during my internship, it didn't calculate the accurate positioning altho it did receive the correction data correctly from FLEPOS.

Reviewer: ```suggestion
There were a couple of problems with the SparkFun GPS-RTK Board during my internship, it didn't calculate the accurate positioning although it did receive the correction data correctly from FLEPOS.
```

The Sparkfun RTK-board is supposed to calculate this automatically, but it didn't and I could not change the configuration to the right settings.
That is why I don't have a video which shows the use of RTK on the SparkFun GPS-RTK Board like in the section with SBAS.

## Sending data to the cloud

> As an extra you can send the data to the cloud, during my internship at Ordina I did this as follows:

Reviewer: If you have more information about this, please expand it. If not, this is fine.

- I received the positioning data with a Raspberry Pi and sent it to the Azure IoT Hub
- Then through Azure functions the data got stored in a Mongodb with Azure Cosmos DB-API
- After that, the data got handled with springboot and visualized.

In the image below you can see the diagram that visualizes this process:
{:refdef: style="text-align: center;"}
<img src="{{ '/img/2022-05-04-RTK-Internship/DiagramRTK.png' | prepend: site.baseurl }}" alt="DiagramRTK" class="image center" style="margin:0px auto; max-width:100%">
{: refdef}

## Conclusion

> Real-Time Kinematic (RTK) can be very accurate and handy in some situations where needed.

Reviewer: The conclusion is a bit small / too little conclusion. Explain in a few sentences what you found (RTK is acurate to x SBAS is accurate to y, small antenna vs good antenna, support for local systems is high)
